services/src/main/java/org/fao/geonet/api/OpenApiController.java

```diff
@@ -180,9 +189,14 @@ protected boolean isRestController(Map<String, Object> restControllers,
                 || !ModelAndView.class.isAssignableFrom(handlerMethod.getMethod().getReturnType()));
     }
 
-    protected void calculateServerUrl(HttpServletRequest request, String apiDocsUrl) {
-        String requestUrl = decode(request.getRequestURL().toString());
-        String calculatedUrl = requestUrl.substring(0, requestUrl.length() - apiDocsUrl.length());
-        this.openAPIService.setServerBaseUrl(calculatedUrl);
+    private String getServerBaseUrl(HttpServletRequest request) {
+        String contextPath = request.getContextPath();
+        StringBuffer requestURL = request.getRequestURL();
```

Comment:
In the previous code it was used with `decode`, was not really required?

Reply:
The old logic did not make any sense to me.
1 - It was using the requestUrl and then subtracting the apiDocsUrl which has no relationship to the current url.
i.e. 
requestUrl was  http://localhost:8080/geonetwork/srv/api/doc
apiDocsUrl was based on @Value(API_DOCS_URL) which was "/v3/api-docs"

It was subtracting the length of "/v3/api-docs" which just happened to be the same length as "/srv/api/doc" which was just luck that this did not fail?

It looked to me like it was just trying to get the base URL and content path from the URL. So the new logic achieves that correctly.

Reply:
@ianwallen thanks for the clarification. Please update the PR to remove the unused imports, it's all fine for me.